connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerSinkTask.java

```diff
@@ -700,9 +714,28 @@ private class HandleRebalance implements ConsumerRebalanceListener {
         @Override
         public void onPartitionsAssigned(Collection<TopicPartition> partitions) {
             log.debug("{} Partitions assigned {}", WorkerSinkTask.this, partitions);
-
+            Set<String> deletedTopics = new HashSet<>();
             for (TopicPartition tp : partitions) {
-                long pos = consumer.position(tp);
+                if (deletedTopics.contains(tp.topic())) {
```

Comment:
I'm not sure I understand why the `onPartitionsAssigned` rebalance listener hook will be called for partitions of topics that are deleted? Is this expected behavior or a bug in the consumer client? If it's expected, is there any documentation around this? If it's a bug, I think it might be better to address the bug in the consumer client rather than introducing a workaround for it in Connect?

Reply:
Yeah that's the behaviour that the consumer client exhibits. I thin it's due to the fact that a topic deletion forces a consumer rebalance and rebalance cycle typically goes through `onPartitionsRevoked` -> `onPArtitionasAssigned` cycle. I am not sure if this is documented anywhere explictily for topic deletion. 
I wouldn't qualify it as a bug though. I think it has been there for a long time. I think Chris had also tried to rope in client team for another suggested change but we haven't heard back yet.

Reply:
Also, this entire block relies on racy logic to detect deleted topics that may fail if a topic is deleted after this check takes place. Again, we should reach out to someone familiar with the clients library. It's unlikely that Kafka Connect is the only consumer application that is impacted by this scenario and it's better to fix this at the clients level instead of implementing Connect-specific workarounds.

Reply:
This block of code uses already existing logic within `TopicAdmin`  to figure out if a topic exists or not. https://github.com/apache/kafka/blob/6d72c26731fe69955127a90e3d43f6d9eb41e2d3/connect/runtime/src/main/java/org/apache/kafka/connect/util/TopicAdmin.java#L503-L511. I am just caching it because if we established that a topic is deleted for partition p1, then we don't need to check again for other partitions of the same topic within this flow.

Reply:
> It's unlikely that Kafka Connect is the only consumer application that is impacted by this scenario

I understand that Connect is not the only consumer application that would be impacted by this behaviour. But ATM, it does impact the users in certain scenarios as described in the ticket. Nonetheless, I would involve somebody from the Client team.